Mastodon/Scene/HomeTimeline/HomeTimelineViewModel+LoadLatestState.swift

```diff
@@ -114,6 +111,22 @@ extension HomeTimelineViewModel.LoadLatestState {
                     if !latestStatusIDs.isEmpty {
                         viewModel.homeTimelineNavigationBarTitleViewModel.newPostsIncoming()
                     }
+                    
+                    var newRecords: [MastodonFeed] = newStatuses.map {
+                        MastodonFeed.fromStatus(.fromEntity($0), kind: .home)
+                    }
+                    viewModel.dataController.records = {
+                        var oldRecords = viewModel.dataController.records
+                        for (i, record) in newRecords.enumerated() {
+                            if let index = oldRecords.firstIndex(where: { $0.status?.reblog?.id == record.id || $0.status?.id == record.id }) {
+                                oldRecords[index] = record
+                                if newRecords.count > index {
+                                    newRecords.remove(at: i)
+                                }
+                            }
+                        }
+                        return (newRecords + oldRecords).removingDuplicates()
+                    }()
```

Comment:
Yes, it's supposed to replace existing status which are returned from the API because they might have changed and additionally to prepend new ones on the timeline.